src/main/java/org/opensearch/ad/feature/CompositeRetriever.java

```diff
@@ -301,12 +337,12 @@ private boolean shouldRetryDueToEmptyPage(SearchResponse response) {
 
         /**
          * Whether next page exists.  Conditions are:
-         * 1) we haven't fetched any page yet (totalResults == 0) or afterKey is not null
+         * 1) this is the first time we query (iterations == 0) or afterKey is not null
          * 2) next detection interval has not started
          * @return true if the iteration has more pages.
          */
         public boolean hasNext() {
-            return (totalResults == 0 || (totalResults > 0 && afterKey != null)) && expirationEpochMs > clock.millis();
+            return (iterations == 0 || (totalResults > 0 && afterKey != null)) && expirationEpochMs > clock.millis();
```

Comment:
Is it possible `iterations == 0` and `afterKey == null`? Not sure if this is enough `afterKey != null && expirationEpochMs > clock.millis()`

Reply:
> Is it possible iterations == 0 and afterKey == null
It is possible. 

> afterKey != null && expirationEpochMs > clock.millis()
This is not enough as at the beginning afterKey = null.